### DISCUSSION TITLE
React - Mapping Through A List And Rendering (With A Custom Component)

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/mern#71] [Tasks]
In this exercise, instead of manually and "imperatively" coding the render of each item in a list, use the map function to "declare" how React should render the list.

Given an array:

['dog', 'cat', 'chicken', 'cow', 'sheep', 'horse']

Use the map function to render them in an unordered list:

- dog
- cat
- chicken
- cow
- horse

Instead of using the `<ul>` and `<li>` HTML tags, create a custom component that accepts a list as its prop, and renders it accordingly.

It's up to you how you want this list will look like. I decided mine to look like this:

![image](https://user-images.githubusercontent.com/60097560/178783372-b3e6d8e4-d663-4485-bf8d-2a5676d9b07d.png)

What's the point of creating custom components? You'll be able to re-use them. Think of it like creating your own functions.

Instead of copy-pasting the same lines of code everywhere you need them, you put those blocks of code inside a separate function, and you just call that function whenever you need it.

This is very good practice because if you need to make changes to your code, you don't need to change it in every place where you used it. You just need to change it in one place.

That's one of the many benefits of using custom components. There are more reasons to using custom components, which needs a whole new blog post, but this is a good start.

**Comment by santu1667** [comment]
```jsx
import './App.css';

function Example(props){
    let animals = props.animals;
    console.log(animals);
    
    return(
        <div className='App'>
            {
                animals.map( animal =>(<p>{animal}</p>)) (// instead of this we can write JS function and call it onLoad)
            }
        </div>
    )
}

export default Example;


 const animals=["dog","cat","horse","meow"]
 Added this route in App-
 <Route exact
          path="/customcomponent"
          element={
            <>
              <Example animals={animals}/>
            </>
          }
        >
</Route>
```

**Comment by sanath8se** [comment]
```jsx
import './App.css';
import Animal from "./Animal.js"
function App() {

  const animals = ['dog', 'cat', 'chicken', 'cow', 'sheep', 'horse'];

  return (
    <div className="App">
      <ul style={{ listStyleType: "none" }}>
        {animals.map((v) => (<Animal v={v}></Animal>))}
      </ul>
    </div>
  );
}

export default App;



function Animal({ v }) {
    return (<li key={v}>{v}</li>);
}

export default Animal;
```

**Reply by sanath8se** [reply]
<img width="288" alt="Screen Shot 2022-07-13 at 12 53 48 PM" src="https://user-images.githubusercontent.com/25965212/178798999-ab9e0655-5a04-45ea-ab6a-812a964f5904.png">

**Comment by santu1667** [comment]
![Screenshot 2022-07-13 125626](https://user-images.githubusercontent.com/106557586/178799586-09f93d1c-9917-45d8-9ea6-3440ef3bd7c6.gif)
